src/components/interaction-designer/block-editors/SimplifiedNameEditor.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+  <div class="simplified-name-editor">
+    <validation-message :message-key="`block/${block.uuid}/name`">
+      <template #input-control="{ isValid }">
+        <div v-if="editBlockName || isValid === false">
+          <text-editor
+            v-model="blockName"
+            :label="''"
+            :placeholder="trans('flow-builder.enter-block-code')"
+            :valid-state="isValid"
+            @keydown="filterName"
+            @mouseleave.native="handleCompleteEditing"/>
+        </div>
+        <div v-else class="block-code">
+          <div @mouseenter="activateEditing"> {{ blockName }} </div>
+        </div>
+      </template>
+    </validation-message>
+  </div>
+</template>
+
+<script lang="ts">
+import {Component, Prop} from 'vue-property-decorator'
+import {mixins} from 'vue-class-component'
+import Lang from '@/lib/filters/lang'
+import {IBlock, IBlockUIMetadata} from '@floip/flow-runner'
+import {namespace} from 'vuex-class'
+
+const flowVuexNamespace = namespace('flow')
+
+@Component({})
+export class SimplifiedNameEditor extends mixins(Lang) {
+  editBlockName = false
+
+  @Prop() readonly block!: IBlock
+  @flowVuexNamespace.Action block_setName!: ({
+    blockId,
+    value,
+    lockAutoUpdate,
+  }: {
+    blockId: IBlock['uuid'],
+    value: IBlock['name'],
+    lockAutoUpdate: boolean,
+  }) => void;
+  @flowVuexNamespace.Action block_resetName!: ({blockId}: { blockId: IBlock['uuid'] }) => void;
+
+  get blockName(): IBlock['name'] {
+    return this.block.name
+  }
+
+  set blockName(value: IBlock['name']) {
+    this.block_setName({
+      blockId: this.block.uuid,
+      value,
+      lockAutoUpdate: true,
+    })
+  }
+
+  updated() {
+    if (this.blockName === '') {
+      this.editBlockName = true
+    }
+  }
+
+  filterName(e: KeyboardEvent): void {
+    this.activateEditing()
+
+    // Consider user input as name editing
```

Comment:
This comment was referring to `activateEditing` line (in the `NameEditor`), let's move it up, or better remove it altogether?

Also, can we add this to the method itself:  `// allow [a-zA-Z0-9_] only`?
Maybe it's just me but I didn't understand at first glance why we were using this _along with_ the regular v-model.

Reply:
@bulatgab , I think the current behaviour does only accept `[a-zA-Z0-9_]` with the below  `e.key.match(/\W+|Enter/g)`, or did I misunderstand your proposal ?

`filterName` is used for `@keydown=`, could you explain a bit the 2nd question?

Reply:
✅ comment cleaned in the recent subtask PR: https://github.com/FLOIP/flow-builder/pull/350/files#diff-1635390784b6b76ab978432bd9c5593ebe899323e62e83332ac0c64f3be6a029L68 (I made it there to avoid cascading updates)

Reply:
comment added as well ;)